File: problems/Valid Word Square/validWordSquare.cs
public class Solution {
    public bool ValidWordSquare(IList<string> words) {
        var rows = words.Count;

        for (var i = 0; rows > i; ++i) {
            for (var j = 0; words[i].Length > j; ++j) {
                if (j >= rows || i >= words[j].Length || words[i][j] != words[j][i]) {
                    return false;
                }
            }
        }

        return true;
    }
}

File: problems/Valid Word Square/validWordSquare.java
class Solution {
    public boolean validWordSquare(List<String> words) {
        var rows = words.size();

        for (int i = 0; rows > i; ++i) {
            for (int j = 0; words.get(i).length() > j; ++j) {
                if (j >= rows || i >= words.get(j).length() || words.get(i).charAt(j) != words.get(j).charAt(i)) {
                    return false;
                }
            }
        }

        return true;
    }
}
